exporter/clickhouseexporter/exporter_metrics_test.go

```diff
@@ -46,93 +46,55 @@ func TestExporter_pushMetricsData(t *testing.T) {
 				if strings.HasPrefix(query, "INSERT INTO otel_metrics_exponential_histogram") {
-					require.Equal(t, "Resource SchemaUrl 1", values[1])
-					require.Equal(t, "Scope name 1", values[2])
-
-					require.Equal(t, "Resource SchemaUrl 2", values[30])
-					require.Equal(t, "Scope name 2", values[31])
+					idx := itemIdxs["otel_metrics_exponential_histogram"]
+					require.Equal(t, fmt.Sprintf("Resource SchemaUrl %d", resourceSchemaIdx[idx.Load()]), values[1])
+					idx.Add(1)
+					require.Equal(t, fmt.Sprintf("Scope name %d", idx.Load()), values[2])
 				}
 				if strings.HasPrefix(query, "INSERT INTO otel_metrics_gauge") {
-					require.Equal(t, "Resource SchemaUrl 1", values[1])
-					require.Equal(t, "Scope name 1", values[2])
-
-					require.Equal(t, "Resource SchemaUrl 2", values[21])
-					require.Equal(t, "Scope name 2", values[22])
+					idx := itemIdxs["otel_metrics_gauge"]
+					require.Equal(t, fmt.Sprintf("Resource SchemaUrl %d", resourceSchemaIdx[idx.Load()]), values[1])
+					idx.Add(1)
+					require.Equal(t, fmt.Sprintf("Scope name %d", idx.Load()), values[2])
 				}
 				if strings.HasPrefix(query, "INSERT INTO otel_metrics_histogram") {
-					require.Equal(t, "Resource SchemaUrl 1", values[1])
-					require.Equal(t, "Scope name 1", values[2])
-
-					require.Equal(t, "Resource SchemaUrl 2", values[26])
-					require.Equal(t, "Scope name 2", values[27])
+					idx := itemIdxs["otel_metrics_histogram"]
+					require.Equal(t, fmt.Sprintf("Resource SchemaUrl %d", resourceSchemaIdx[idx.Load()]), values[1])
+					idx.Add(1)
+					require.Equal(t, fmt.Sprintf("Scope name %d", idx.Load()), values[2])
 				}
 				if strings.HasPrefix(query, "INSERT INTO otel_metrics_sum (") {
-					require.Equal(t, "Resource SchemaUrl 1", values[1])
-					require.Equal(t, "Scope name 1", values[2])
-
-					require.Equal(t, "Resource SchemaUrl 2", values[23])
-					require.Equal(t, "Scope name 2", values[24])
+					idx := itemIdxs["otel_metrics_sum"]
+					require.Equal(t, fmt.Sprintf("Resource SchemaUrl %d", resourceSchemaIdx[idx.Load()]), values[1])
+					idx.Add(1)
+					require.Equal(t, fmt.Sprintf("Scope name %d", idx.Load()), values[2])
 				}
 				if strings.HasPrefix(query, "INSERT INTO otel_metrics_summary") {
-					require.Equal(t, "Resource SchemaUrl 1", values[1])
-					require.Equal(t, "Scope name 1", values[2])
-
-					require.Equal(t, "Resource SchemaUrl 2", values[19])
-					require.Equal(t, "Scope name 2", values[20])
+					idx := itemIdxs["otel_metrics_summary"]
+					require.Equal(t, fmt.Sprintf("Resource SchemaUrl %d", resourceSchemaIdx[idx.Load()]), values[1])
+					idx.Add(1)
+					require.Equal(t, fmt.Sprintf("Scope name %d", idx.Load()), values[2])
 				}
 			}
 			return nil
 		})
 		exporter := newTestMetricsExporter(t)
 		mustPushMetricsData(t, exporter, simpleMetrics(1))
 
-		require.Equal(t, int32(5), items.Load())
-	})
-	t.Run("check one resource metrics contain different scope metrics", func(t *testing.T) {
```

Comment:
I think it's worth keeping it(it confirm this [bug](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21209) is fixed )

Reply:
The test case is removed because it is already included in the previous one. The updated test case "check Resource metadata and scope metadata" checked all the generated "Resource SchemaUrl"s and "Scope name"s.

Reply:
I still think we will lose unit test coverage: a resource metrics could have different scope metrics, as you can see: 
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9e17996df1db3271512fc043386a553803b2fbcb/exporter/clickhouseexporter/exporter_metrics_test.go#L376-L382

The `simpleMetrics()` mocks the second resource metrics with two different scope metrics, current unit tests don't cover it.

Reply:
I have refacted the the case to be readable. Previously using an index to represent the 'scope name' and 'resource schemaUrl' id.

```
resourceSchemaIdx := []int{1, 2, 2} // 1st resource with 1 scope metric, and 2nd resource with 2 scope metrics.
scopeNameIdx := []int{1, 2, 3}        // 3 scope metrics with increase index.
```